examples/YOLOv8-SAHI-Inference-Video/YOLOv8_SAHI_ObjectTracking.py

```diff
@@ -0,0 +1,193 @@
+import cv2
+import cvzone
+import argparse
+import numpy as np
+from pathlib import Path
+from sahi import AutoDetectionModel
+from deep_sort.deep_sort import DeepSort
+from deep_sort.sort.tracker import Tracker
+from deep_sort.utils.parser import get_config
+from sahi.predict import get_sliced_prediction
+from ultralytics.utils.files import increment_path
+from sahi.utils.yolov8 import download_yolov8s_model
+
+
+def run(
+    weights="yolov8n.pt",
+    source="test.mp4",
+    without_tracking=False,
+    view_img=False,
+    save_img=False,
+    exist_ok=False,
+    tracking=False,
+):
+    """
+    Run object detection on a video using YOLOv8 and SAHI.
+
+    Args:
+        weights (str): Model weights path.
+        source (str): Video file path.
+        without_tracking (bool): Disable DeepSort tracking.
+        view_img (bool): Show results.
+        save_img (bool): Save results.
+        exist_ok (bool): Overwrite existing files.
+        tracking (bool): Enable DeepSort tracking.
+    """
+
+    # Check source path
+    is_webcam = source.isdigit() and int(source) in range(10)
+    if not is_webcam and not Path(source).exists():
+        raise FileNotFoundError(f"Source path '{source}' does not exist.")
+
+    if is_webcam:
+        source = int(source)
+
+    yolov8_model_path = f"models/{weights}"
+    download_yolov8s_model(yolov8_model_path)
+    detection_model = AutoDetectionModel.from_pretrained(
+        model_type="yolov8", model_path=yolov8_model_path, confidence_threshold=0.3, device="cuda"
+    )
+
+    # Video setup
+    videocapture = cv2.VideoCapture(source)
+    frame_width, frame_height = int(videocapture.get(3)), int(videocapture.get(4))
+    fps, fourcc = int(videocapture.get(5)), cv2.VideoWriter_fourcc(*"mp4v")
+
+    # Output setup
+    save_dir = increment_path(Path("ultralytics_results_with_sahi") / "exp", exist_ok)
+    save_dir.mkdir(parents=True, exist_ok=True)
+
+    if is_webcam:
+        video_writer = cv2.VideoWriter(
+            str(save_dir / f"{str(source)}_output.mp4"), fourcc, fps, (frame_width, frame_height)
+        )
+    else:
+        video_writer = cv2.VideoWriter(
+            str(save_dir / f"{Path(source).stem}.mp4"), fourcc, fps, (frame_width, frame_height)
+        )
+
+    tracker = None
+    if tracking:
+        config = get_config()
+        tracker = DeepSort(model_path="deep_sort/deep/checkpoint/ckpt.t7", max_age=70)
+
+    while videocapture.isOpened():
+        success, frame = videocapture.read()
+        if not success:
+            break
+
+        results = get_sliced_prediction(
+            frame, detection_model, slice_height=512, slice_width=512, overlap_height_ratio=0.2, overlap_width_ratio=0.2

```

Comment:
@MuhammadBilal848 i would suggest setting slice_height and slice_width proportional to the frame resolution. In other words setting them dynamically as slice_height = frame_height/zoom_factor, slice_width = frame_width/zoom_factor where zoom_factor is 1.5 if user wants to apply 1.5x zoom.

Moreover, overlap_height_ratio and overlap_width_ratio can be set as 0.05 or 0.1 for faster inference

Reply:
I tried making the changes locally that you asked but they are raising errors. Specifically **slice_height = frame_height//zoom_factor, slice_width = frame_width//zoom_factor**. So I'm removing the changes.

Reply:
@MuhammadBilal848 can you please paste the full error traceback so that I can try to come up with a solution?

Reply:
Amazing fix @MuhammadBilal848 !